### PR TITLE
chore: add a label sync tools

### PR DESCRIPTION
**Pull request by triclambert**
Following instructions from here:  https://github.com/opstrace/opstrace/blob/b00d2a20701f2455f1340bb1b23a5de9b85011ca/ci/github-issue-labels.md

* [x] A label config file change is proposed with a pull request and discussed among us
* [x] As part of the pull request we show the dry run output.
* [ ] After merge, we run the actual sync and update the PR with the output.

```
$ ./node_modules/.bin/github-label-sync  --dry-run --labels opstrace-labels.yaml opstrace/opstrace
zsh: no such file or directory: ./node_modules/.bin/github-label-sync

$ yarn add -W github-label-sync
...
success Saved lockfile.
success Saved 6 new dependencies.
info Direct dependencies
└─ github-label-sync@2.0.2
info All dependencies
├─ @financial-times/origami-service-makefile@7.0.3
├─ array-uniq@1.0.2
├─ github-label-sync@2.0.2
├─ node.extend@2.0.2
├─ octonode@0.10.2
└─ randomstring@1.2.1
Done in 29.64s.

$ ./node_modules/.bin/github-label-sync  --dry-run --labels opstrace-labels.yaml opstrace/opstrace
No labels were found in /home/ubuntu/opstrace/opstrace-labels.yaml

$ ./node_modules/.bin/github-label-sync  --dry-run --labels ci/github-issue-labels.yaml opstrace/opstrace
No labels were found in /home/ubuntu/opstrace/ci/github-issue-labels.yaml

$ wc -l /home/ubuntu/opstrace/ci/github-issue-labels.yaml
147 /home/ubuntu/opstrace/ci/github-issue-labels.yaml
```

¯\\_(ツ)_/¯

### CODE REVIEW

**Comment by triclambert** [comment]
Interesting:  I cannot repair the DCO on this PR.  The initial commit was made with GitHub's new Codespaces feature.  I wonder why the instructions are not working (I have used the many times before).

**Comment by jgehrcke** [comment]
I am sure the DCO check can be fixed.

Here is what I did to get to the dry run output:

```
$ yarn add -W github-label-sync
$ ./node_modules/.bin/github-label-sync --dry-run --labels ci/github-issue-labels.yaml opstrace/opstrace
Syncing labels for "opstrace/opstrace"
Fetching labels from GitHub
 > Changed: the "state: backlog" label in the repo is out of date. It will be updated to "backlog" with color "#dddddd" and description "state (used by codetree)".
 > Changed: the "state: in-progress" label in the repo is out of date. It will be updated to "in progress" with color "#dddddd" and description "state (used by codetree)".
 > Changed: the "state: wontdo" label in the repo is out of date. It will be updated to "state: wontdo" with color "#dddddd".
 > Changed: the "state: invalid" label in the repo is out of date. It will be updated to "state: invalid" with color "#dddddd".
 > Added: the "in progress" label in the repo is not expected. It will be deleted.
 > Added: the "backlog" label in the repo is not expected. It will be deleted.
 > Added: the "codetree-epic" label in the repo is not expected. It will be deleted.
 > Added: the "dependencies" label in the repo is not expected. It will be deleted.
 > Added: the "area: test-browser" label in the repo is not expected. It will be deleted.
```

hat was _without_ any changes to `ci/github-issue-labels.yaml`. that implies a number of TODO items (easy).

> chore: add action to auto-label PRs

hm. Let's discuss this separately. I am skeptical :).

**Comment by triclambert** [comment]
> Here is what I did to get to the dry run output:

Did you run that from this branch?  How do you run the tool before?

I guess you can execute the change at the end.

> I am sure the DCO check can be fixed.

Any ideas?

> hm. Let's discuss this separately. I am skeptical :).

At the moment all that's there is adding the docs label for changes to the docs directory.  Which seems reasonable to me.

**Comment by jgehrcke** [comment]
> all that's there is adding the docs label for changes to the docs directory

I had seen that. But for what is that useful / which kind of problem does this help with?

Will every PR in which I happen to _also_ touch docs get a `docs` label?

Will someone sometimes filter for "docs" PRs to .... answer which question?

**Comment by jgehrcke** [comment]
> Did you run that from this branch?

from `main`. 

> Any ideas? 

Rebase, add sign-off to each commit.

**Comment by triclambert** [comment]
> Rebase, add sign-off to each commit.

Great, that worked.  Thanks.

**Comment by triclambert** [comment]
> Will every PR in which I happen to also touch docs get a docs label?

Yup.

> Will someone sometimes filter for "docs" PRs to .... answer which question?

Something along the lines of, _show me all of the "docs" PRs so that I can review them for some reason._

My thinking here is that we should use the `area: docs` label more than we are today or remove it.  Since few people seem to apply it manually, we can fix that by adding it automatically.

**Comment by triclambert** [comment]
@jgehrcke redirected this PR to be about label tools.  Specifically, adding github-labels-sync to our project and the auto-labeling of docs PRs.  Perhaps we can discuss these here?

File: package.json [hunk 1]
```diff
@@ -50,6 +50,8 @@
     "wsrun": "^5.2.4"
   },
   "dependencies": {
+    "@types/js-yaml": "^3.12.5",
+    "github-label-sync": "^2.0.2",
```
Comment (jgehrcke) [inline package.json:54]: quick note: should be part of devDependencies if at all, not in `dependencies` (as you can see we've been working pretty hard to make this list 'empty' here -- can elaborate on why that is in sync conv :)).

File: commitlint.config.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ module.exports = {
       "always",
       [
         "ui", // use `ui: ` for changes to client, and `ui: app:` for changes to server
+        "alerts", // use `alerts: ` for changes to alert rules, definitions, thresholds, etc.
```
Comment (jgehrcke) [inline commitlint.config.js:25]: :+1:

File: package.json (hunk 1 above)
Comment (triclambert) [reply package.json]: Ok, no problem.  Updated.

**Comment by triclambert** [comment]
Closing as won't do, for now.

**Pull request closed without merging**